src/jit/lowerxarch.cpp

```diff
@@ -2409,14 +2409,29 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
         }
     }
-    else if (intrinsicID == NI_SSE_Shuffle) // TODO - change to all IMM intrinsics
+
+    if (Compiler::categoryOfHWIntrinsic(intrinsicID) == HW_Category_IMM)
     {
-        assert(op1->OperIsList());
-        GenTree* op3 = op1->AsArgList()->Rest()->Rest()->Current();
+        int numArgs = Compiler::numArgsOfHWIntrinsic(intrinsicID);
+        assert(numArgs >= 2);
+        GenTree* lastOp = nullptr;
+        if (numArgs == 2)
+        {
+            lastOp = op2;
+        }
+        else
+        {
+            assert(numArgs == 3);
+            assert(op1->OperIsList());
+            lastOp = op1->AsArgList()->Rest()->Rest()->Current();
+        }
 
-        if (op3->IsCnsIntOrI())
+        if (Compiler::isImmHWIntrinsic(intrinsicID, lastOp))
```

Comment:
A comment explaining why we do this check in addition to the category check above would be useful